### DISCUSSION TITLE
Load pandas function

### DISCUSSION THREAD

**Discussion by EricDallAgnol** [atoti/atoti#757] [Q&A] [answered]
Hi team,

I have a quick question, maybe naïve, regarding the [load_pandas](https://docs.atoti.io/latest/lib/atoti/atoti.table.html#atoti.Table.load_pandas) function. 

If I provide as argument to this function a df that has more columns than expected in my target table, will them be filtered out before being sent to AP ?

_For instance, let's assume I have a df with COL_1, COL_2 and COL_3 but in my target table I have only COL_1 and COL_2.
Inside AP I know COL_3 will be ignored, but will Atoti filter out this one first before being sent or will it be only handled by the underlying AP process ?_

Many thanks,
Eric

**Comment by tibdex** [comment, marked as answer]
Hi Eric, `Table.load_pandas()` converts the pandas DataFrame to an Arrow table and then calls `Table.load_arrow()` which writes the data to a file then read by the server (AP). The extras columns are not removed in the Python side.

Are you asking this because you're having performance issues?

**Reply by EricDallAgnol** [reply]
Hi Thibault,

Thanks for your answer !
No I'm not facing performance issues. 
We are reworking/improving our ETL layer and we have one file parsed that is used to populate 2 tables with different columns and we wanted to understand if we have to handle the removal of the extra-columns before calling _load_pandas_  😃

**Reply by tibdex** [reply]
Hi Eric, just a heads-up: there's been an unintentional change in 0.7.3 (released yesterday) related to this: https://github.com/atoti/atoti/issues/758. You can subscribe to that issue to follow progress on this.

**Reply by EricDallAgnol** [reply]
Hi Thibault,

Ok good to know, many thanks for the update !
